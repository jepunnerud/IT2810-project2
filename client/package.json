{
  "name": "prosjekt-2",
  "private": true,
  "version": "0.0.0",
  "type": "module",
  "scripts": {
    "dev": "vite",
    "build": "tsc && vite build",
    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
    "preview": "vite preview",
    "prettier": "prettier . --write",
    "lint:prettier": "prettier . --check",
    "test": "vitest",
    "cypress": "cypress open"
  },
  "dependencies": {
    "@apollo/client": "^3.8.7",
    "@testing-library/react": "^14.1.2",
    "graphql": "^16.8.1",
    "prettier": "^3.0.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-hook-form": "^7.48.2",
    "react-router-dom": "^6.16.0"
  },
  "devDependencies": {
    "@types/react": "^18.2.15",
    "@types/react-dom": "^18.2.7",
    "@typescript-eslint/eslint-plugin": "^6.0.0",
    "@typescript-eslint/parser": "^6.0.0",
    "@vitejs/plugin-react": "^4.0.3",
    "cypress": "^13.5.1",
    "eslint": "^8.45.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.3",
    "jsdom": "^22.1.0",
    "typescript": "^5.0.2",
    "vite": "^4.4.5",
    "vite-test-utils": "^0.6.0",
    "vitest": "^0.34.6"
  }
}
