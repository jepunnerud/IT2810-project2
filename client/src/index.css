:root {
  --background-color: #c7f9cc;
  --primary-color: #80ed99;
  --secondary-color: #57cc99;
  --tertiary-color: #38a3a5;
  --quaternary-color: #22577a;

  --dark-background-color: #151515;
  --dark-primary-color: #2a2727;
  --dark-secondary-color: #393433;
  --dark-tertiary-color: #80ed99;
  --dark-quaternary-color: #c7f9cc;

  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
  line-height: 1.5;
  font-weight: 400;

  color-scheme: light dark;
  color: black;
  background-color: var(--background-color);
  font-synthesis: none;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-size-adjust: 100%;
}

a {
  text-decoration: inherit;
}

body {
  overflow: scroll;
}

.button {
  user-select: none;
  background-color: var(--primary-color);

  &.dark {
    background-color: var(--dark-primary-color);
    color: white;
  }

  &.disabled {
    background-color: gray !important;
    opacity: 0.4;
    cursor: not-allowed !important;
  }

  color: black;
  border-radius: 1rem;
  margin: 0.5rem;
  border: none;

  &:hover {
    cursor: pointer;

    &:not(:active) {
      background-color: var(--secondary-color);

      &.dark {
        background-color: var(--dark-secondary-color);
      }
    }
  }
}
