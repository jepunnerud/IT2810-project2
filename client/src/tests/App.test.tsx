import { render, waitFor, screen } from '@testing-library/react'
import HomePage from '../pages/Home'
import { describe, expect, it } from 'vitest'
import { MockedProvider } from '@apollo/client/testing'
import { gql } from '@apollo/client'
import { BrowserRouter } from 'react-router-dom'

describe('Home Page snapshot test', () => {
  beforeEach(() => {
    vi.mock('react-router-dom', async () => {
      const actual = (await vi.importActual('react-router-dom')) as object
      return {
        ...actual,
        useSearchParams: () => [new URLSearchParams()],
        useNavigate: () => vi.fn(),
      }
    })
  })

  const mocks = [
    {
      request: {
        query: gql`
          query GetAllDrinks($ingredient: String, $limit: Int, $skip: Int, $sort: String) {
            drinks(ingredient: $ingredient, limit: $limit, skip: $skip, sort: $sort) {
              drinks {
                _id
                name
                picture
              }
              pageInfo {
                totalCount
                totalPages
              }
            }
          }
        `,
        variables: {
          ingredient: '',
          limit: 12,
          skip: 0,
          sort: 'name-asc',
        },
      },
      result: {
        data: {
          drinks: {
            drinks: [
              {
                _id: '1',
                name: 'TestDrink',
                picture: 'https://www.thecocktaildb.com/images/media/drink/5noda61589575158.jpg',
                ingredients: [
                  {
                    ingredient: 'Test Ingredient',
                  },
                ],
              },
            ],
            pageInfo: { totalCount: 1, totalPages: 1 },
          },
        },
      },
    },
  ]
  const renderHomePage = () => {
    render(
      <MockedProvider addTypename={false} mocks={mocks}>
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </MockedProvider>
    )
  }

  it('should render the home page', async () => {
    const homePage = renderHomePage()
    await waitFor(() => {
      screen.getAllByTestId('drink-card-TestDrink')
    })
    expect(homePage).toMatchSnapshot()
  })
})
