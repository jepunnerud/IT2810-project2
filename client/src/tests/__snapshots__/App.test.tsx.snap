// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Page snapshot test > should render the home page 1`] = `undefined`;
